app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -243,6 +169,129 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
+        if (isTablet == this.isTablet) return
+
+        if (isTablet) {
+            addTabletActions(requireContext())
+        } else {
+            removeTabletActions()
+        }
+
+        this.isTablet = isTablet
+    }
+
+    @Suppress("LongMethod")
+    private fun addTabletActions(context: Context) {
+        val enableTint = ThemeManager.resolveAttribute(R.attr.textPrimary, context)
+        val disableTint = ThemeManager.resolveAttribute(R.attr.textDisabled, context)
+        backAction?.let {
+            browserToolbarView.view.addNavigationAction(it)
+        } ?: kotlin.run {
+            backAction = BrowserToolbar.TwoStateButton(
+                primaryImage = AppCompatResources.getDrawable(
+                    context,
+                    R.drawable.mozac_ic_back,
+                )!!,
+                primaryContentDescription = context.getString(R.string.browser_menu_back),
+                primaryImageTintResource = enableTint,
+                isInPrimaryState = { getCurrentTab()?.content?.canGoBack ?: false },
+                secondaryImageTintResource = disableTint,
+                disableInSecondaryState = true,
+                longClickListener = {
+                    browserToolbarInteractor.onBrowserToolbarMenuItemTapped(
+                        ToolbarMenu.Item.Back(viewHistory = true),
+                    )
+                },
+                listener = {
+                    browserToolbarInteractor.onBrowserToolbarMenuItemTapped(
+                        ToolbarMenu.Item.Back(viewHistory = false),
+                    )
+                },
+            ).also {
+                browserToolbarView.view.addNavigationAction(it)
+            }
+        }
+
+        forwardAction?.let {
+            browserToolbarView.view.addNavigationAction(it)
+        } ?: kotlin.run {
+            forwardAction = BrowserToolbar.TwoStateButton(
+                primaryImage = AppCompatResources.getDrawable(
+                    context,
+                    R.drawable.mozac_ic_forward,
+                )!!,
+                primaryContentDescription = context.getString(R.string.browser_menu_forward),
+                primaryImageTintResource = enableTint,
+                isInPrimaryState = { getCurrentTab()?.content?.canGoForward ?: false },
+                secondaryImageTintResource = disableTint,
+                disableInSecondaryState = true,
+                longClickListener = {
+                    browserToolbarInteractor.onBrowserToolbarMenuItemTapped(
+                        ToolbarMenu.Item.Forward(viewHistory = true),
+                    )
+                },
+                listener = {
+                    browserToolbarInteractor.onBrowserToolbarMenuItemTapped(
+                        ToolbarMenu.Item.Forward(viewHistory = false),
+                    )
+                },
+            ).also {
+                browserToolbarView.view.addNavigationAction(it)
```

Comment:
**[Open Question]** It's a bit odd to see that we have to repeat `browserToolbarView.view.addNavigationAction(it)` twice in these blocks.
I am wondering if it would just be simpler to just do the following:

```
if (forwardAction == null) {
  forwardAction = BrowserToolbar.TwoStateButton(...)
}

forwardAction?.let {
    browserToolbarView.view.addNavigationAction(it)
}
```

Reply:
ya, I agree it's better to just leave it a simple null check and set